src/legacy/core_plugins/console/public/src/helpers/settings_show_modal.tsx

```diff
@@ -54,12 +54,22 @@ export function showSettingsModal() {
     prevSettings: DevToolsSettings
   ) => {
     // We'll only retrieve settings if polling is on.
-    const isPollingChanged = prevSettings.polling !== newSettings.polling;
     if (newSettings.polling) {
       const autocompleteDiff = getAutocompleteDiff(newSettings, prevSettings);
-      if (autocompleteDiff.length > 0) {
-        mappings.retrieveAutoCompleteInfo(newSettings.autocomplete);
-      } else if (isPollingChanged) {
+
+      const isSettingsChanged = autocompleteDiff.length > 0;
+      const isPollingChanged = prevSettings.polling !== newSettings.polling;
+
+      if (isSettingsChanged) {
+        // If the user has changed one of the autocomplete settings, then we'll fetch just the
+        // ones which have changed.
+        const changedSettings = autocompleteDiff.reduce((changedSettingsAccum, setting) => {
+          changedSettingsAccum[setting] = newSettings.autocomplete[setting];
+          return changedSettingsAccum;
+        }, {});
+        mappings.retrieveAutoCompleteInfo(changedSettings);
+      } else if (isPollingChanged && newSettings.polling) {
```

Comment:
~Is `&& newSettings.polling` needed? Looks like it is already being checked on line `57`.~

After looking at this further, should the initial check on line `57` be removed? I think we would want to fetch any changed settings regardless if polling is turned on or off.

Reply:
My line of thought here is that users disabled polling because the fetch request is so expensive. So, I think they'd be unhappy with any fetch that's automatically executed without them clicking the "Fetch now" button. Does that make sense? I'm going to add a comment to explain that for now but I'm still open to changing this if you think we can improve it.

Reply:
Ok, makes sense. Thanks for adding the comment! I _think_ you can still delete `&& newSettings.polling` (my original comment). Otherwise, this is fine with me.

Reply:
`isPollingChanged` will also be `true` if the user has turned off polling. So we need to make sure we're fetching the settings only if the user has turned polling on, but not when they've turned it off.

Reply:
isn't that already being handled on line 60 though?
```
if (newSettings.polling) { ... 
```

Reply:
Oh you're right! I missed that. Thank you for persisting in pointing it out to me 😅  Will fix!